text/0036-auto-reference-docs.md

```diff
@@ -0,0 +1,188 @@
+implementation of cloud-native buildpacks are unique and don't fit the
+requirements for some existing automation tools since not all the documentation
+will be dependent directly on the code or code comments. Designing a new
+automation tool gives us the ability to fine tune how we want to display the
+reference documentation on the Paketo website. It will also give maintainers
+the ability to provide supplemental documentation if they want.
+
+## Rationale and Alternatives
+
+There are two ways to add reference documentation to the Paketo website:
+1. Have maintainers create PRs to the Paketo website whenever there is a change
+   to their respective buildpacks.
+2. Create a tool that aggregates the documentation from the individual
+   buildpack themselves automatically whenever there is a new change.
+
+The first option makes it harder to maintain up-to-date documentation as this
+would add an extra step for maintainers whenever they make a change to the
+buildpack. With the second option, maintainers can update the docs within the
+same commit as the changes.
+
+## Implementation
+
+In order to achieve automation we are proposing the creation of a new CLI tool
+or the extension of an existing CLI tool (i.e. `jam` or `libpak/cmd`). This
+tool will consume the structures in order to generate reference documentation
+in the form of markdown. The first structure we are proposing to consume is the
+`buildpack.toml` with inclusion of the `metadata.configurations` table array.
+Examples of this field can be seen in the Java buildpacks such as [BellSoft Liberica](https://github.com/paketo-buildpacks/bellsoft-liberica):
+```
+api = "0.6"
+
+[buildpack]
+id       = "paketo-buildpacks/bellsoft-liberica"
+
+...
+
+[[metadata.configurations]]
+name        = "BPL_JVM_HEAD_ROOM"
+description = "the headroom in memory calculation"
+default     = "0"
+launch      = true
+
+...
+
+[[metadata.configurations]]
+name        = "BP_JVM_TYPE"
+description = "the JVM type - JDK or JRE"
+default     = "JRE"
+build       = true
+
+...
+
+[[metadata.dependencies]]
+id      = "jdk"
+name    = "BellSoft Liberica JDK"
+version = "8.0.302"
+uri     = "https://github.com/bell-sw/Liberica/releases/download/8u302+8/bellsoft-jdk8u302+8-linux-amd64.tar.gz"
+sha256  = "23628d2945e54fc9c013a538d8902cfd371ff12ac57df390869e492002999418"
+stacks  = [ "io.buildpacks.stacks.bionic", "io.paketo.stacks.tiny", "*" ]
+
+  [[metadata.dependencies.licenses]]
+  type = "GPL-2.0 WITH Classpath-exception-2.0"
+  uri  = "https://openjdk.java.net/legal/gplv2+ce.html"
+
+...
+```
+
+We propose that all maintainers add this configurations table array to their
+buildpacks as a way of automatically generating information about the
+environment variables that can be configured to modify the buildpacks'
+behavior.
+
+Additional documentation can be generated using an optional file included in
+the project's root named `reference-doc.md`. The `reference-doc.md` file would
```

Comment:
I feel like this location should be flexible. I might want to keep this under `docs/reference-doc.md` or `docs/reference.md`. I think the proposal should allow for that.

Reply:
I am more than happy to allow for alternate location for the `reference-doc.md` I am also not necessarily tied to its currently location.

Reply:
OK. My vote would be for `docs/reference.md`, or something like `.docs.yml` that is like `reference-doc-path: other/reference-path.md`.